tests/rados/test_online_reads_balancer.py

```diff
@@ -0,0 +1,168 @@
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    pool_obj = PoolFunctions(node=cephadm)
+
+    regex = r"\s*(\d.\d)-rhel-\d"
+    build = (re.search(regex, config.get("build", config.get("rhbuild")))).groups()[0]
+    if not float(build) >= 8.0:
+        log.info(
+            "Test running on version less than 7.0, skipping verifying Reads Balancer functionality"
+        )
+        return 0
+
+    try:
+        log.info(
+            "Starting the test to verify Online reads balancer with Balncer module "
+        )
+        log.debug("Creating multiple pools for testing")
+        time.sleep(10)
+        pools = config["create_pools"]
+        for each_pool in pools:
+            cr_pool = each_pool["create_pool"]
+            if cr_pool.get("pool_type", "replicated") == "erasure":
+                method_should_succeed(
+                    rados_obj.create_erasure_pool, name=cr_pool["pool_name"], **cr_pool
+                )
+            else:
+                method_should_succeed(rados_obj.create_pool, **cr_pool)
+            method_should_succeed(rados_obj.bench_write, **cr_pool)
+        log.info("Completed creating pools and writing test data into the pools")
+
+        log.debug(
+            "Checking the scores on each pool on the cluster, if the pool is replicated pool"
+        )
+        read_scores_pre = rados_obj.get_read_scores_on_cluster()
+        log.info(
+            f"Completed collection of balance scores for all the pools before enabling Online reads balancer"
+            f"Scores are : {read_scores_pre}"
+        )
+
+        # Tracker : https://tracker.ceph.com/issues/66274 .
+        # -ve scenario . We should not be able to move to upmap-read or read profiles without setting min-compat-client
+        try:
+            rados_obj.enable_balancer(balancer_mode="upmap-read")
+        except Exception as err:
+            log.info(
+                f"Hit expected exception on the cluster. Error : {err}"
+                f"Checking currently enabled mode on the cluster"
+            )
+            cmd = "ceph balancer status"
+            balancer_status = rados_obj.run_ceph_command(cmd=cmd)
+            if balancer_status["mode"] == "upmap-read":
+                log.error(
+                    f"Balancer mode updated without setting min-compat-client on the cluster."
+                    f"balancer status : {balancer_status}"
+                )
+                raise Exception("Balancer upmap-read mode should not be enabled Error")
+            log.info("Verified balancer mode. Could not be enabled. Proceeding")
+
+        log.debug(
+            "Setting config to allow clients to perform read balancing on the clusters."
+        )
+        config_cmd = "ceph osd set-require-min-compat-client reef"
+        rados_obj.run_ceph_command(cmd=config_cmd)
+        time.sleep(10)
+
+        if not rados_obj.enable_balancer(balancer_mode="upmap-read"):
```

Comment:
can we also validate "read" mode alone? along with "upmap-read"? May be in next PR and add backlog tracker.

Reply:
Made the changes. Made the balancer mode a configurable param from the suite file.